src/Microsoft.AspNet.OData.Shared/DeltaOfTStructuralType.cs

```diff
@@ -736,6 +782,60 @@ private void CopyChangedDynamicValues(TStructuralType targetEntity)
                 targetEntity);
         }
 
+        /// <summary>
+        /// Copies changed nested properties and leaves the unchanged properties
+        /// </summary>
+        /// <param name="original">The structural object</param>
+        private void CopyChangedNestedProperties(TStructuralType original)
+        {
+            // For nested resources.
+            foreach (string nestedResourceName in _deltaNestedResources.Keys)
+            {
+                // Patch for each nested resource changed under this TStructuralType.
+                dynamic deltaNestedResource = _deltaNestedResources[nestedResourceName];
+                dynamic originalNestedResource = null;
+
+                if (!TryGetPropertyRef(original, nestedResourceName, out originalNestedResource))
+                {
+                    throw Error.Argument(nestedResourceName, SRResources.DeltaNestedResourceNameNotFound,
+                        nestedResourceName, original.GetType());
+                }
+
+                if (originalNestedResource == null)
+                {
+                    // When patching original target of null value, directly set nested resource.
+                    dynamic deltaObject = _deltaNestedResources[nestedResourceName];
```

Comment:
Why do we need to initialize this variable here yet `deltaNestedResource` initialized on [Line 795](https://github.com/OData/WebApi/pull/2686/files#diff-ed3f5d87f1b8b1557c0c4d21e427b43e8e7858520e990c2610bb1ab620d3d727R795) has the same value?